js/editor-ta.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * @constructor
+ * @param {DOM} elementId
+ * @param {Settings} settings
+ */
+function EditorTextArea(editorElement, settings) {
+  this.element_ = editorElement;
+  this.settings_ = settings;
+  this.numLines_ = 1;
+
+  this.setTheme();
+
+  const initFontSize = this.settings_.get('fontsize') + 'px';
+
+  // The main text area the user will be interacting with
+  this.textarea_ = document.createElement('textarea');
+  this.textarea_.setAttribute('id', 'editor-ta');
+  this.textarea_.style.fontSize = initFontSize;
+  this.textarea_.addEventListener('input', function() {
+    this.onChange();
+  }.bind(this));
+  window.addEventListener('resize', function() {
+    this.onResize();
+  }.bind(this));
+
+  this.wrapper_ = document.createElement('div');
+  this.wrapper_.classList.add('editor-ta-wrapper');
+  this.wrapper_.appendChild(this.textarea_);
+
+  this.lineNumbers_ = document.createElement('div');
+  this.lineNumbers_.setAttribute('id', 'editor-ta-line-numbers');
+  this.lineNumbers_.style.fontSize = initFontSize;
+  this.lineNumbers_.appendChild(this.lineNumberElem(1));
+
+  this.container_ = document.createElement('div');
+  this.container_.classList.add('editor-ta-container');
+  this.container_.append(this.lineNumbers_);
+  this.container_.append(this.wrapper_);
+
+  this.containerResizeObserver_ = new ResizeObserver(function() {
+    this.onResize();
+  }.bind(this)).observe(this.container_);
+
+  this.element_.appendChild(this.container_);
+
+  // A hack which leverages a hidden div to figure out the height and width
```

Comment:
not really, if you do the math with just number of lines and font size it consistently comes out too small or too big depending on font size. Most answers online seem to suggest this is the most stable way to do it

Reply:
height and width are important because the textarea needs to match the height of all the text + the width of the largest line so if users flip word wrap off the text area can be the correct with to let users scroll horizontally.

Reply:
Interesting, do you know why it the calc comes out wrong? Maybe add a code comment about that approach not working

Reply:
oh! In researching why this happens it turns out the issue is that variable line heights cause issues as it adds space above and below a line, but since we are explicitly setting a line height we can do the math ourselves, will try update the code :)

Reply:
So turns out that the height of the text is off because we were not considering line height, i  can update the calculations but there is still the issue of having to figure out the width and height of the text box given word wrap. I think this method is a good way to handle that case since writing our own code to figure out when the line wraps is a bit annoying.
That being said i will tweek the code to use calculations where possible (i.,e when word wrap is off)

Reply:
sgtm